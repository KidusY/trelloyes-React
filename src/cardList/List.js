import React from 'react'
import './List.css';
import Card from '../Card/Card.js';




function List(props) {
    const getCards= props.cardIds.map(id=><Card key={id} id={props.id} card={props.allCards[id]} onDelete={props.onDelete}/>)
    return(
        <div className='List'>
            <header className='list-header'>
               <h2>{props.header}</h2>
            </header>
            <button className='List-add-button' type='button' onClick={()=>props.onAdd(props.id)}>Add random card</button>
            <div className='List-cards'>
                {getCards}
            </div>
        </div>
    )    
}
List.defaultProps = {
    cardIds: [],
    allCards: {'': {  }},
    header: ''
}

export default List;