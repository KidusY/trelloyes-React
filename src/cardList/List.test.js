import React from 'react'
import ReactDOM from 'react-dom'
import List from './List'
import renderer from 'react-test-renderer'

it ('I just wanted to do the smoke test', () => {
    const div = document.createElement('div')
    ReactDOM.render(<List />, div)
    ReactDOM.unmountComponentAtNode(div)
})

it ('I just wanted to do the snapshot test', () => {
    const myRenderedElement1= renderer.create(<List />).toJSON();
    expect(myRenderedElement1).toMatchSnapshot();
})
