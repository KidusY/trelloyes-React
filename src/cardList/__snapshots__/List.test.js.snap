// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`I just wanted to do the snapshot test 1`] = `
<div
  className="List"
>
  <header>
    <h2>
       
      
       
    </h2>
  </header>
  <div
    className="List-cards"
  />
</div>
`;
